src/server/database/Database/Implementation/WorldDatabase.cpp

```diff
@@ -24,8 +24,11 @@ void WorldDatabaseConnection::DoPrepareStatements()
         m_stmts.resize(MAX_WORLDDATABASE_STATEMENTS);
 
     PrepareStatement(WORLD_SEL_QUEST_POOLS, "SELECT entry, pool_entry FROM pool_quest", CONNECTION_SYNCH);
-    PrepareStatement(WORLD_DEL_CRELINKED_RESPAWN, "DELETE FROM linked_respawn WHERE guid = ?", CONNECTION_ASYNC);
-    PrepareStatement(WORLD_REP_CREATURE_LINKED_RESPAWN, "REPLACE INTO linked_respawn (guid, linkedGuid) VALUES (?, ?)", CONNECTION_ASYNC);
+    PrepareStatement(WORLD_DEL_CREATURE_LINKED_RESPAWN, "DELETE FROM linked_respawn WHERE guid = ? AND linkType IN (0,1)", CONNECTION_ASYNC); //CREATURE_TO_CREATURE, CREATURE_TO_GO
```

Comment:
i would hide the values of the linkType from db layer and fill the values when you use the prepared queries

Reply:
Make sense

Reply:
Would you prefer, for example in the case of  WORLD_DEL_CREATURE_LINKED_RESPAWN:
- Always feeding two values of CreatureLinkedRespawnType to the same query. But that doesn't look very generic to me.
- Calling the query two times to remove both CREATURE_TO_CREATURE and CREATURE_TO_GO types. This also implies we have no less than four queries just to remove a creature, since we also must add two for the *_MASTER queries.

Maybe go with the second option but make it a function that takes a reference to a transaction to avoid cluttering the DeleteFromDB functions with 4 queries ?

Reply:
Would you prefer, for example in the case of WORLD_DEL_CREATURE_LINKED_RESPAWN:

Always feeding two values of CreatureLinkedRespawnType to the same query. But that doesn't look very generic to me.
Calling the query two times to remove both CREATURE_TO_CREATURE and CREATURE_TO_GO types.
I'd go with the latter option but this also implies we have no less than four queries just to remove a creature, since we also must add two for the *_MASTER queries.
Maybe go with the latter option but make it a RemoveLinkedRespawn function that takes a reference to a transaction to avoid cluttering the DeleteFromDB functions with 4 queries ?

Reply:
Would you prefer, for example in the case of WORLD_DEL_CREATURE_LINKED_RESPAWN:

Always feeding two values of CreatureLinkedRespawnType to the same query. But that doesn't look very generic to me.
Calling the query two times to remove both CREATURE_TO_CREATURE and CREATURE_TO_GO types.
I'd go with the latter option but this also implies we have no less than four queries just to remove a creature, since we also must add two for the *_MASTER queries.
Maybe go with the latter option but make it a RemoveLinkedRespawn function that takes a reference to a transaction to avoid cluttering the DeleteFromDB functions with 4 queries ?

Reply:
```sql
DELETE FROM linked_respawn WHERE guid = ? AND linkType = ?
```
that way would be fine
don't care about the 4 queries, it is the job of the dbms